### ISSUE TITLE
Does not check Link references

### ISSUE THREAD

**Issue by jcalcaben** [trevorblades/gatsby-remark-check-links#4]
This currently does not work with reference link.

## Example

```md
[Broken link][]

[Broken link]: /foo
```

## Expected result

Plugin reports a broken link

## Actual

Plugin does not report a broken link.

## Proposed solution

Create a new visitor for [Definition](https://github.com/syntax-tree/mdast#definition) nodes to test for those locations.
